Sources/Extensions/UIView+PinLayout.swift

```diff
@@ -95,4 +102,42 @@ extension UIView: Layoutable, SizeCalculable {
     }
 }
 
+extension UIView: AutoSizeCalculable {
+    private struct pinlayoutAssociatedKeys {
+        static var pinlayoutAutoSizingRect = UnsafeMutablePointer<Int8>.allocate(capacity: 1)
+        static var pinlayoutAutoSizingRectWithMargins = UnsafeMutablePointer<Int8>.allocate(capacity: 1)
+    }
+
+    public var autoSizingRect: CGRect? {
+        get {
+            return objc_getAssociatedObject(self, &pinlayoutAssociatedKeys.pinlayoutAutoSizingRect) as? CGRect
+        }
+        set {
+            objc_setAssociatedObject(self, &pinlayoutAssociatedKeys.pinlayoutAutoSizingRect, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    public var autoSizingRectWithMargins: CGRect? {
+        get {
+            return objc_getAssociatedObject(self, &pinlayoutAssociatedKeys.pinlayoutAutoSizingRectWithMargins) as? CGRect
+        }
+        set {
+            objc_setAssociatedObject(self, &pinlayoutAssociatedKeys.pinlayoutAutoSizingRectWithMargins, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    public func autoSizeThatFits(_ size: CGSize, layoutClosure: () -> Void) -> CGSize {
+        Pin.autoSizingInProgress = true
```

Comment:
Probably not a good idea because if you have multiple nested custom views that all use `autoSizeThatFits` they will all try to mutate the global flag but that's an expected behaviour in that case.